api/app/controllers/spree/api/orders_controller.rb

```diff
@@ -14,6 +14,12 @@ class OrdersController < Spree::Api::BaseController
         end
       end
 
+      def approve
+        authorize! :update, @order, params[:token]
```

Comment:
@reinaris This can be achieved with a new custom action so `authorize! :approve, @order`

By default admin has the ability to manage all objects and therefore would be able to approve this but user's can only update so will not.